.circleci/config.yml

```
@@ -85,6 +91,10 @@ references:
         # run steps in CircleCI, so we specify the full path to pre-commit
         # explicitly here.
         python${PYVER} -m pre_commit run --all-files
+        # We need to generate the compile_commands.json database, which
+        # requires the build.
+        python${PYVER} setup.py build_ext --inplace -- -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -- -j 4
+        "`python${PYVER} -m site --user-base`/bin/pre-commit" run -c .pre-commit-config-cpp.yaml --all-files
```

Use `python -m pre_commit` here as in #685.
```suggestion
        python -m pre_commit run -c .pre-commit-config-cpp.yaml --all-files
```